sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -427,8 +427,12 @@ case class MapObjects private(
       case _ => ""
     }
 
+    val inputDataType = inputData.dataType match {
+      case p: PythonUserDefinedType => p.sqlType
```

Comment:
This probably needs a comment explaining the reasoning behind the code. Also, why are we allowed to use p.sqlType for inputDataType serialization in the case of a python UDT? The UDT's deserialize() still has to be called for each element of the array, right?